src/Models/IoTContainerizedProject.ts

```diff
@@ -23,142 +24,104 @@ const raspberryPiDeviceModule =
-      projectType: ProjectTemplateType, boardId: string,
-      openInNewWindow: boolean) {
-    // Step 0: Check prerequisite
+      templateFilesInfo: TemplateFileInfo[], projectType: ProjectTemplateType,
+      boardId: string, openInNewWindow: boolean): Promise<void> {
     // Can only create project locally
     await RemoteExtension.checkRemoteExtension();
 
     const createTimeScaffoldType = ScaffoldType.Local;
-    if (rootFolderPath !== undefined) {
+
+    // Create project root path
+    if (!await FileUtility.directoryExists(
+            createTimeScaffoldType, this.projectRootPath)) {
       await FileUtility.mkdirRecursively(
-          createTimeScaffoldType, rootFolderPath);
-    } else {
-      throw new Error(
-          'Unable to find the root path, please open the folder and initialize project again.');
+          createTimeScaffoldType, this.projectRootPath);
     }
 
-    this.projectRootPath = rootFolderPath;
+    // Update iot workbench project file
+    await this.updateIotWorkbenchProjectFile(createTimeScaffoldType);
 
-    await this.generateOrUpdateIotWorkbenchProjectFile(
-        createTimeScaffoldType, this.projectRootPath);
-
-    const iotworkbenchprojectFile =
-        path.join(this.projectRootPath, FileNames.iotworkbenchprojectFileName);
-    if (!await FileUtility.fileExists(
-            createTimeScaffoldType, iotworkbenchprojectFile)) {
-      throw new Error(
-          `Internal Error. Could not find iot workbench project file.`);
-    }
-
-    let projectConfig: {[key: string]: string} = {};
-    const projectConfigContent =
-        (await FileUtility.readFile(
-             createTimeScaffoldType, iotworkbenchprojectFile, 'utf8') as string)
-            .trim();
-    if (projectConfigContent) {
-      projectConfig = JSON.parse(projectConfigContent);
-    }
+    const projectConfig = await this.getProjectConfig(createTimeScaffoldType);
 
     // Step 1: Create device
-    let device;
-    if (boardId === raspberryPiDeviceModule.RaspberryPiDevice.boardId) {
-      device = new raspberryPiDeviceModule.RaspberryPiDevice(
-          this.extensionContext, this.projectRootPath, this.channel,
-          this.telemetryContext, templateFilesInfo);
-    } else {
-      throw new Error('The specified board is not supported.');
-    }
-
+    await this.initDevice(boardId, createTimeScaffoldType, templateFilesInfo);
     projectConfig[`${ConfigKey.boardId}`] = boardId;
 
+    // Update workspace config to workspace config file
+    if (!this.iotWorkbenchProjectFilePath) {
+      throw new Error(
+          `Workspace config file path is empty. Please initialize the project first.`);
+    }
+    await FileUtility.writeJsonFile(
+        createTimeScaffoldType, this.iotWorkbenchProjectFilePath,
+        projectConfig);
+
+    // Check components prerequisites
+    this.componentList.forEach(async item => {
+      const res = await item.checkPrerequisites();
+      if (!res) {

```

Comment:
do we need to throw error if prerequisites is not met?

Reply:
Yes. Fixed.